

<!doctype html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Notes on Zygocity &#8212; Anfisa Development&amp;Installation Documentation v.0.6 documentation</title>
    <link rel="stylesheet" href="_static/bizstyle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Inventory (*.CFG) file format" href="adm/a_inventory.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0">
    <!--[if lt IE 9]>
    <script type="text/javascript" src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="adm/a_inventory.html" title="Inventory (*.CFG) file format"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Anfisa Development&amp;Installation Documentation v.0.6 documentation</a> &#187;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Notes on Zygocity</a><ul>
<li><a class="reference internal" href="#definitions-case-samples">Definitions: case, samples</a></li>
<li><a class="reference internal" href="#zygocity-states">Zygocity states</a></li>
<li><a class="reference internal" href="#zygocity-conditions">Zygocity conditions</a></li>
<li><a class="reference internal" href="#zygocity-scenario">Zygocity scenario</a></li>
<li><a class="reference internal" href="#standard-zygocity-scenarios">Standard zygocity scenarios</a></li>
<li><a class="reference internal" href="#zygocity-events-and-compound-requests">Zygocity events and compound requests</a></li>
<li><a class="reference internal" href="#compound-heterozygous-variants">Compound heterozygous variants</a></li>
<li><a class="reference internal" href="#approximations-to-gene-locations">Approximations to gene locations</a></li>
<li><a class="reference internal" href="#conclusion">Conclusion</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="adm/a_inventory.html"
                        title="previous chapter">Inventory (*.CFG) file format</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/zygocity.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="notes-on-zygocity">
<h1>Notes on Zygocity<a class="headerlink" href="#notes-on-zygocity" title="Permalink to this headline">¶</a></h1>
<p>The purpose of this text is to explain (or interpret) some portion of genomics in terms of
informatics. We take care of correct informational formulations for data structures
(in JSON format) and tasks to be supported.</p>
<div class="section" id="definitions-case-samples">
<h2>Definitions: case, samples<a class="headerlink" href="#definitions-case-samples" title="Permalink to this headline">¶</a></h2>
<p><a class="reference internal" href="glossary.html#term-dataset"><span class="xref std std-term">Dataset</span></a> represents <strong>case</strong> i.e genomics information on one or many persons - <strong>samples</strong> -
from the same family. Each sample in case is identified by <strong>id</strong>.</p>
<p>Examples of id: <code class="docutils literal notranslate"><span class="pre">&quot;bgm9001a1&quot;</span></code>, <code class="docutils literal notranslate"><span class="pre">&quot;bgm9001u1&quot;</span></code></p>
<p>Sample id starts with case id, then letter <code class="docutils literal notranslate"><span class="pre">a</span></code> (affected) or <code class="docutils literal notranslate"><span class="pre">u</span></code> (unaffected), and a number
from 1 to upper. Id of proband always ends up with <code class="docutils literal notranslate"><span class="pre">&quot;a1&quot;</span></code> (in medical cases we proband is always “affected”) .</p>
<p>The typical case is <strong>trio</strong>: samples are proband and two parents. Note that some tasks in below
are formulated only in context of this scope.</p>
</div>
<div class="section" id="zygocity-states">
<h2>Zygocity states<a class="headerlink" href="#zygocity-states" title="Permalink to this headline">¶</a></h2>
<p>Most of chromosomes for human are pairwise, so a mutation/variant applying to a sample has
one of the following states:</p>
<blockquote>
<div><ul class="simple">
<li>Zygocity 0: variant is absent</li>
<li>Zygocity 1: variant is heterozygous, it presents on one chromosome copy</li>
<li>Zygocity 2: variant is homozygous, it presents on both copies of chromosome</li>
</ul>
</div></blockquote>
<p>Males have chromosomes X and Y in one copies, so a mutation/variant on these chromosomes
for male sample has one of the following states:</p>
<blockquote>
<div><ul class="simple">
<li>Zygocity 0: variant is absent</li>
<li>Zygocity 2: variant is hemozygous, it does present on single copy of chromosome</li>
</ul>
</div></blockquote>
<p><em>Attention</em>: in hemozygous state a variant presents in one copy, however it is more convenient
to define zygocity number as 2, since it is essentially more similar to homozygous case than
to heterozygous one.</p>
</div>
<div class="section" id="zygocity-conditions">
<h2>Zygocity conditions<a class="headerlink" href="#zygocity-conditions" title="Permalink to this headline">¶</a></h2>
<p>We need to define meaningful conditions for zygocity number for variants.
So let us define the following 5 <em>string</em> (not <em>int</em>!) constants:</p>
<p><code class="docutils literal notranslate"><span class="pre">“2”</span></code>,    <code class="docutils literal notranslate"><span class="pre">“2-1”</span></code>,    <code class="docutils literal notranslate"><span class="pre">“1”</span></code>,    <code class="docutils literal notranslate"><span class="pre">“1-0”</span></code>,    <code class="docutils literal notranslate"><span class="pre">“0”</span></code></p>
<p>Formally speaking, we can define one more meaningful constant: <code class="docutils literal notranslate"><span class="pre">&quot;0-2&quot;</span></code>, but let us interpret
this case as we do not define condition at all: all variant are passed.</p>
</div>
<div class="section" id="zygocity-scenario">
<span id="id1"></span><h2>Zygocity scenario<a class="headerlink" href="#zygocity-scenario" title="Permalink to this headline">¶</a></h2>
<p>Zygocity scenario is join of zygocity conditions for samples in case. In JSON format
it has form:</p>
<blockquote>
<div><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">{</span></code> <em>dictionary</em></div>
<div class="line-block">
<div class="line">&lt;zygocity condition constant&gt;: <code class="docutils literal notranslate"><span class="pre">[</span></code> <em>list of</em> samples id <code class="docutils literal notranslate"><span class="pre">]</span></code></div>
<div class="line">…</div>
</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">}</span></code></div>
</div>
</div></blockquote>
</div>
<div class="section" id="standard-zygocity-scenarios">
<span id="id2"></span><h2>Standard zygocity scenarios<a class="headerlink" href="#standard-zygocity-scenarios" title="Permalink to this headline">¶</a></h2>
<p>Let us consider split of samples in case on two groups: affected/unaffected.</p>
<p>The following scenarios are important in practice:</p>
<blockquote>
<div><ul class="simple">
<li><strong>Homozygous Recessive/X-linked</strong>:  <code class="docutils literal notranslate"><span class="pre">{“2”:</span></code> <em>affected</em>, <code class="docutils literal notranslate"><span class="pre">“1-0”:</span></code> <em>unaffected</em> <code class="docutils literal notranslate"><span class="pre">}</span></code></li>
<li><strong>Autosomal Dominant</strong>:             <code class="docutils literal notranslate"><span class="pre">{“2-1”:</span></code> <em>affected</em>, <code class="docutils literal notranslate"><span class="pre">“0”:</span></code> <em>unaffected</em> <code class="docutils literal notranslate"><span class="pre">}</span></code></li>
<li><strong>Compensational</strong>:                 <code class="docutils literal notranslate"><span class="pre">{“0”:</span></code> <em>affected</em>, <code class="docutils literal notranslate"><span class="pre">“2-1”:</span></code> <em>unaffected</em> <code class="docutils literal notranslate"><span class="pre">}</span></code></li>
</ul>
</div></blockquote>
<p>Comments:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>Autosomal Dominant</strong> - probably bad: variant presents for all affected and is absent for</dt>
<dd>all unaffected</dd>
<dt><strong>Homozygous Recessive</strong> - variant probably breaks gene, but has no effect while</dt>
<dd>there is second good (gene) copy</dd>
<dt><strong>X-linked</strong> - variant probably bad, it affects all male samples and females</dt>
<dd>in homozygous state</dd>
<dt><strong>Compensational</strong> - variant is probably good, it fixes (screens) up illness caused</dt>
<dd>by other facts</dd>
</dl>
</div></blockquote>
<p>Notes:</p>
<blockquote>
<div><ul class="simple">
<li>we call group of affected samples as <strong>problem group</strong></li>
<li><dl class="first docutils">
<dt>for a dataset the <strong>default problem group</strong> is preset by different</dt>
<dd>letters <code class="docutils literal notranslate"><span class="pre">a</span></code> and <code class="docutils literal notranslate"><span class="pre">u</span></code> in id of samples;
however it is good if the user can define problem group directly</dd>
</dl>
</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="zygocity-events-and-compound-requests">
<span id="compound-request"></span><h2>Zygocity events and compound requests<a class="headerlink" href="#zygocity-events-and-compound-requests" title="Permalink to this headline">¶</a></h2>
<p>Let us fix some diapason or diapasons of positions on chromosome as “gene” (see
<a class="reference internal" href="#gene-aproximations"><span class="std std-ref">below</span></a> for details). Then for some
fixed zygocity scenario it is possible to count how many variants with this scenario presents
inside the “gene”. Let us call these variants as “events”.</p>
<blockquote>
<div><p>We are ready to define <strong>compound request</strong> in the following form:</p>
<div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">[</span></code> <em>list of</em> conditions for scenarios</div>
<div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">[</span></code> <em>list</em></div>
<div class="line-block">
<div class="line"><strong>[0]</strong>: minimal count of events, <strong>positive int</strong></div>
<div class="line"><strong>[1]</strong>: <a class="reference internal" href="#zygocity-scenario"><span class="std std-ref">zygocity scenario</span></a></div>
</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">]</span></code>, …</div>
</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">]</span></code></div>
</div>
</div></blockquote>
<p>The result of compound request is not empty only if for all scenarios in the list there are not less than the defined count of events.</p>
<p>Strongly speaking, a compound request applies to “genes”, and its result is either empty, or
<em>list of lists</em> of variants that cause events. The current version of the system does not support
so complex selection objects, so we interpret result of compound request as a joined plain list of variants.</p>
</div>
<div class="section" id="compound-heterozygous-variants">
<span id="compound-heterozygous"></span><h2>Compound heterozygous variants<a class="headerlink" href="#compound-heterozygous-variants" title="Permalink to this headline">¶</a></h2>
<p>The following compound request is important in practice. It applies to trio:</p>
<blockquote>
<div><div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">[</span></code></div>
<div class="line-block">
<div class="line"><code class="docutils literal notranslate"><span class="pre">[1,</span> <span class="pre">{“1”:</span> <span class="pre">{</span></code> <em>proband</em>, <em>proband’s mother</em> <code class="docutils literal notranslate"><span class="pre">},</span> <span class="pre">“0”:</span> <span class="pre">{</span></code> <em>proband`s father</em> <code class="docutils literal notranslate"><span class="pre">}},</span></code></div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">[1,</span> <span class="pre">{“1”:</span> <span class="pre">{</span></code> <em>proband</em>, <em>proband’s father</em> <code class="docutils literal notranslate"><span class="pre">},</span> <span class="pre">“0”:</span> <span class="pre">{</span></code> <em>proband`s mother</em> <code class="docutils literal notranslate"><span class="pre">}}</span></code></div>
</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">]</span></code></div>
</div>
</div></blockquote>
<p>Comments.</p>
<ul class="simple">
<li><dl class="first docutils">
<dt>The main idea of request: let us consider case when only proband is affected. It might</dt>
<dd>happen that on some “gene” each parent has (different) dangerous variant in heterozygous state.
These variants have no effect because second copy of gene are not broken. But proband has both
copies of “gene” broken: one copy is broken by one of dangerous variants, ans second copy -
by another one.</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>From informational point of view, the detection of compound events is a clear tasks.</dt>
<dd>In practice however there are serious difficulties to prepare proper setup for this
detection. Results of the procedure might be good only if most part of benign variants
are filtered out before the detection process. It is a matter of experiments, so the
system provides extended functionality for this special kind of experimental activity.</dd>
</dl>
</li>
</ul>
</div>
<div class="section" id="approximations-to-gene-locations">
<span id="gene-aproximations"></span><h2>Approximations to gene locations<a class="headerlink" href="#approximations-to-gene-locations" title="Permalink to this headline">¶</a></h2>
<p>The system supports 3 variants of gene approximation:</p>
<blockquote>
<div><ul class="simple">
<li><code class="docutils literal notranslate"><span class="pre">&quot;transcript&quot;</span></code>: shared transcript</li>
<li><code class="docutils literal notranslate"><span class="pre">&quot;gene&quot;</span></code>: shared gene</li>
<li><code class="docutils literal notranslate"><span class="pre">&quot;rough&quot;</span></code>: non-intersecting transcripts</li>
</ul>
</div></blockquote>
<p>The first two variants use <a class="reference internal" href="glossary.html#term-transcript"><span class="xref std std-term">transcripts</span></a> as a base filtering item, so
they are applicable only in <a class="reference internal" href="glossary.html#term-ws-dataset"><span class="xref std std-term">WS-datasets</span></a>.</p>
<p>In practice the first variant is most good for precision purposes. But it might be not so
good in recall: not all transcripts are well studied and registered up to now. For recall
purposes use <code class="docutils literal notranslate"><span class="pre">&quot;rough&quot;</span></code> approximation: it causes many false positive effects however
it filters out variants that can not be found with first two approximation variants.</p>
</div>
<div class="section" id="conclusion">
<h2>Conclusion<a class="headerlink" href="#conclusion" title="Permalink to this headline">¶</a></h2>
<p>Detection for variants of standard scenarios and compound heterozygous variants are standard tasks
in genomics, so it is important to support it in the most easy way.</p>
<p>On another hand, the functionality based on direct definition of scenarios and/or compound requests
is rather heavy for support and usage. But it is important, especially in complex cases, with
many samples in case.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="adm/a_inventory.html" title="Inventory (*.CFG) file format"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Anfisa Development&amp;Installation Documentation v.0.6 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright Copyright (c) 2019, 2020. Partners HealthCare and other members of Forome Association.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.4.
    </div>
  </body>
</html>